stingray/crossspectrum.py

```diff
@@ -210,49 +209,112 @@ def rebin(self, df, method="mean"):
         """
         Normalize the real part of the cross spectrum to Leahy, absolute rms^2,
         fractional rms^2 normalization, or not at all.
 
         Parameters
         ----------
-        unnorm_cs: numpy.ndarray
+        unnorm_power: numpy.ndarray
             The unnormalized cross spectrum.
 
         tseg: int
             The length of the Fourier segment, in seconds.
 
         Returns
         -------
-        cs: numpy.nd.array
+        power: numpy.nd.array
             The normalized co-spectrum (real part of the cross spectrum). For
             'none' normalization, imaginary part is returned as well.
         """
 
-        ## The "effective" count rate is the geometrical mean of the count rates
-        ## of the two light curves; need to divide by tseg to have count rate
-        actual_mean = np.sqrt(self.nphots1 * self.nphots2 / tseg)
+        # The "effective" counst/bin is the geometrical mean of the counts/bin
+        # of the two light curves
+
+        log_nphots1 = np.log(self.nphots1)
+        log_nphots2 = np.log(self.nphots2)
+
+        actual_nphots = np.float64(np.sqrt(np.exp(log_nphots1 + log_nphots2)))
+        actual_mean = np.sqrt(self.meancounts1 * self.meancounts2)
 
         assert actual_mean > 0.0, \
                 "Mean count rate is <= 0. Something went wrong."
 
         if self.norm.lower() == 'leahy':
-            c = unnorm_cs.real
-            cs = c * 2. / actual_mean
+            c = unnorm_power.real
+            power = c * 2. / actual_nphots
 
         elif self.norm.lower() == 'frac':
-            c = unnorm_cs.real / np.float(self.n**2.)
-            cs = c * 2. * tseg / (actual_mean**2.0)
+            c = unnorm_power.real / np.float(self.n**2.)
+            power = c * 2. * tseg / (actual_mean**2.0)
 
         elif self.norm.lower() == 'abs':
-            c = unnorm_cs.real / np.float(self.n**2.)
-            cs = c * (2. * tseg)
+            c = unnorm_power.real / np.float(self.n**2.)
+            power = c * (2. * tseg)
 
         elif self.norm.lower() == 'none':
-            cs = unnorm_cs
+            power = unnorm_power
+
+        else:
+            raise Exception("Normalization not recognized!")
+
+        return power
+
+    def rebin_log(self, f=0.01):
+        """
+        Logarithmic rebin of the periodogram.
+        The new frequency depends on the previous frequency
+        modified by a factor f:
+
+        dnu_j = dnu_{j-1}*(1+f)
+
+        Parameters
+        ----------
+        f: float, optional, default 0.01
+            parameter that steers the frequency resolution
+
+
+        Returns
+        -------
+        binfreq: numpy.ndarray
+            the binned frequencies
+
+        binpower: numpy.ndarray
+            the binned powers
+
+        nsamples: numpy.ndarray
+            the samples of the original periodogramincluded in each
+            frequency bin
+        """
+
+        minfreq = self.freq[1] * 0.5  # frequency to start from
```

Comment:
Since `_fourrier_cross` returns `freqs[freqs>0]` this means that `self.freq[0] == self.df` and `self.freq[1] == 2*df`.
This means that `self.freq[1]*0.5 == self.freq[0]` this line should be changed to:

> minfreq = self.freq[0]

or 

> minfreq = self.freq[0] * 0.5

Depending on what is the expected behavior (I believe it's the second case, since in the end we shift the bin edges by `0.5*df`)